### PR TITLE
added python 3.6 specific stuff

### PR DESCRIPTION
**Pull request by tarekziade**
This patche would allow me to run the conditioned profile task on android devices. see https://bugzilla.mozilla.org/show_bug.cgi?id=1562870

I am using a modern Python version along with a couple of python packages, on requires 3.6

I don't think this change will affect the current docker image in any particular way, besides the fact that it will have 3.6 installed.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,6 +3,17 @@ FROM ubuntu:16.04
...
 RUN add-apt-repository ppa:openjdk-r/ppa
 
+# upgrading to Python 3.6
+RUN apt-get install curl
+RUN add-apt-repository ppa:jonathonf/python-3.6
+RUN apt-get update
+RUN apt-get install -y python3.6
+RUN curl https://bootstrap.pypa.io/get-pip.py | python3.6
+
+# adding a couple of packages for the conditioned profiles
```
Comment (aerickson) [inline Dockerfile]: I think the test framework should install pips.

Changing this image is a very heavyweight process. We don't want to start installing testing deps here.

File: Dockerfile [hunk 2]
```diff
@@ -3,6 +3,17 @@ FROM ubuntu:16.04
 RUN apt-get update && apt-get install -y software-properties-common python-software-properties
 RUN add-apt-repository ppa:openjdk-r/ppa
 
+# upgrading to Python 3.6
+RUN apt-get install curl
+RUN add-apt-repository ppa:jonathonf/python-3.6
```
Comment (aerickson) [inline Dockerfile:4]: Python 3 and pip for it is available in the official/default repos, there's no need for a PPA for them.

```
python3
python3-pip
```

**Review by bclary — CHANGES_REQUESTED** [review]
A few comments...

* We already have curl.
* You should add the repository above where we add the openjdk.
* Try not to create so many intermediate images with the separate RUN commands. You should be able to include some of these installs in the existing RUN... apt-get install
* We need to also version this change and test it.
* I'm not sure about the best approaching for getting python 3.6 in these images. @aerickson can talk with bitbar to see what they say.

@aerickson should have the final say here.

**Comment by bclary** [comment]
PS. The curl and pip commands should go in the final RUN command as well.

**Review by aerickson — COMMENTED** [review]
From reading the bug it seems like you want to run on an emulated devices. This Dockerfile is only used by android hardware devices.

File: Dockerfile (hunk 1 above)
Comment (tarekziade) [reply Dockerfile]: We can't vendor python 3 only packages at this point (that should be possible in the future but not now) 
I understand that changing this Dockerfile is a very heavyweight process, but why adding two more package changes anything ? the Dockerfile installs mozdevice for instance.

File: Dockerfile (hunk 2 above)
Comment (tarekziade) [reply Dockerfile:4]: This version of Ubuntu comes with Python 3.5, this installs Python 3.6

**Comment by tarekziade** [comment]
@aerickson using the emulator was the first intent because I did not want the extra complexity to run things on bitbar. But I reached a blocker using emulators, as we want to run the script on arm as well as x86. android *arm* emulators are unusable from a x86 device (like 100x slower) and often crash. 

I understand from your feedback that changing this Dockerfile is a pain, but I don't have any other way. Is there anything I can do to help getting this done ? Happy to go through that heavy weight process or at least help.

**Review by tarekziade — COMMENTED** [review]
@bclary reviews done

**Comment by tarekziade** [comment]
supersed by https://github.com/bclary/mozilla-bitbar-docker/pull/12

**Pull request closed without merging**